modules/api/src/main/java/org/eclipse/xpanse/api/controllers/ServiceRecreateApi.java

```diff
@@ -112,74 +113,51 @@ public UUID recreateService(@Valid @PathVariable("serviceId") String serviceId)
-            @Parameter(name = "recreateStatus", description = "Status of the service recreate")
-            @RequestParam(name = "recreateStatus", required = false)
-            RecreateStatus recreateStatus
-    ) {
-        return recreateService.listServiceRecreates(UUID.fromString(recreateId),
-                UUID.fromString(serviceId),
-                recreateStatus, getUserId());
+        return new ServiceOrder(updatedRecreateOrderEntity.getOrderId(),
+                updatedRecreateOrderEntity.getOriginalServiceId());
     }
 
-    /**
-     * Get recreate records based on recreate id.
-     *
-     * @param recreateId ID of the service recreate.
-     * @return serviceRecreateEntity.
-     */
-    @Tag(name = "Service", description = "APIs to manage the services")
-    @Operation(description = "Get recreate records based on recreate id.")
-    @GetMapping(value = "/services/recreate/{recreateId}", produces =
-            MediaType.APPLICATION_JSON_VALUE)
-    @ResponseStatus(HttpStatus.OK)
-    @AuditApiRequest(methodName = "getCspFromServiceRecreateId")
-    public ServiceRecreateDetails getRecreateOrderDetailsById(
-            @Parameter(name = "recreateId", description = "Recreate ID")
-            @PathVariable("recreateId") String recreateId) {
-        return recreateService.getRecreateOrderDetails(UUID.fromString(recreateId), getUserId());
+    private DeployTask getRecreateTask(ServiceDeploymentEntity serviceDeploymentEntity) {
+        DeployTask recreateTask = new DeployTask();
+        recreateTask.setOrderId(UUID.randomUUID());
+        recreateTask.setTaskType(ServiceOrderType.RECREATE);
+        recreateTask.setServiceId(serviceDeploymentEntity.getId());
+        recreateTask.setOriginalServiceId(serviceDeploymentEntity.getId());
+        recreateTask.setUserId(getUserId());
+        recreateTask.setRequest(serviceDeploymentEntity.getDeployRequest());
+        return recreateTask;
     }
 
     private String getUserId() {
-        return userServiceHelper.getCurrentUserId();
+        return this.userServiceHelper.getCurrentUserId();
     }
 
     private Map<String, Object> getRecreateProcessVariable(
             ServiceDeploymentEntity deployedService,
-            String userId) {
+            ServiceOrderEntity recreateOrderEntity) {
         Map<String, Object> variable = new HashMap<>();
-        variable.put(RecreateConstants.ID, deployedService.getDeployRequest().getServiceId());
+        variable.put(RecreateConstants.SERVICE_ID, deployedService.getId());
+        variable.put(RecreateConstants.RECREATE_ORDER_ID, recreateOrderEntity.getOrderId());
         variable.put(RecreateConstants.RECREATE_REQUEST, deployedService.getDeployRequest());
-        variable.put(RecreateConstants.USER_ID, userId);
+        variable.put(RecreateConstants.USER_ID, recreateOrderEntity.getUserId());
+        variable.put(RecreateConstants.DEPLOY_RETRY_NUM, 0);
+        variable.put(RecreateConstants.DESTROY_RETRY_NUM, 0);
```

Comment:
I think we should add a constant named `RERATE_ERROR_MESSAGE` RecreateConstants, which uesd to store the error_message in the filed deploy order or destory order.